### PR TITLE
Add pasrer to logger

### PR DESCRIPTION
**Pull request by Hipst0man**

### CODE REVIEW

File: CPPLogger/CPPLogger.h [hunk 1]
```diff
@@ -22,31 +22,37 @@ class ILogger
...
 	Logger& operator<<(const char* log_message) override;
 
-	void set_filter_level(LogLevels level) override;
+	void set_filter_level(unsigned int level) override;
 
 private:
+	Logger();
+	Logger(const Logger&);
+	
+	static Logger*  instance;
```
Comment (CharlesLyleLaRue) [inline CPPLogger/CPPLogger.h:44]: For static members name should be:
s_instance;

File: CPPLogger/CPPLogger.vcxproj [hunk 2]
```diff
@@ -91,7 +91,7 @@
       <PreprocessorDefinitions>WIN32;_DEBUG;_LIB;%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <ConformanceMode>true</ConformanceMode>
       <PrecompiledHeaderFile>pch.h</PrecompiledHeaderFile>
-      <AdditionalIncludeDirectories>%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
+      <AdditionalIncludeDirectories>C:\Users\ymasytc\source\repos\SMTP-Client-Server\CPPLogger\include\p7Headers;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
```
Comment (CharlesLyleLaRue) [inline CPPLogger/CPPLogger.vcxproj:94]: Path should be specified as relative.

File: CPPLogger/CPPLogger.vcxproj [hunk 3]
```diff
@@ -115,15 +115,15 @@
       <PreprocessorDefinitions>_DEBUG;_LIB;%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <ConformanceMode>true</ConformanceMode>
       <PrecompiledHeaderFile>pch.h</PrecompiledHeaderFile>
-      <AdditionalIncludeDirectories>%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
+      <AdditionalIncludeDirectories>C:\Users\ymasytc\source\repos\SMTP-Client-Server\CPPLogger\include\p7Headers;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
```
Comment (CharlesLyleLaRue) [inline CPPLogger/CPPLogger.vcxproj:118]: Should be relative path.

**Review by CharlesLyleLaRue — COMMENTED** [review]
Check comments.

File: CPPLogger/CPPLogger.cpp [hunk 4]
```diff
@@ -63,7 +74,17 @@ Logger& Logger::operator<<(const char* log_message)
...
 }
 
-void Logger::set_filter_level(LogLevels level)
+void Logger::set_filter_level(unsigned int level)
 {
+	if (level < 0 || level > 5) return;
 	m_trace->Set_Verbosity(I_HMODULE, (eP7Trace_Level)level);
 }
+
+Logger* Logger::GetInstance()
```
Comment (Stepbura) [inline CPPLogger/CPPLogger.cpp:83]: Please use Double-Checked Locking Singleton

File: XMLParser/XMLParser.cpp [hunk 5]
```diff
@@ -26,7 +26,7 @@ bool XMLParser::ParseFile(const std::string& filename)
 	std::ifstream file{ filename };
 	if (!file.is_open())
 	{
-		LOG_INFO << "Can't open file.";
+		//LOG_INFO << "Can't open file.";
```
Comment (Stepbura) [inline XMLParser/XMLParser.cpp:29]: remove commented code

File: XMLParser/XMLParser.h [hunk 6]
```diff
@@ -94,5 +94,5 @@ class XMLParser final : public IXMLParser
 
 private:
 	std::shared_ptr<XMLNode> m_root;
-	Logger LOG;
+	//Logger LOG;
```
Comment (Stepbura) [inline XMLParser/XMLParser.h:97]: remove commented code

File: CPPLogger/CPPLogger.h (hunk 1 above)
Comment (Hipst0man) [reply CPPLogger/CPPLogger.h:44]: Fixed

File: XMLParser/XMLParser.cpp (hunk 5 above)
Comment (Hipst0man) [reply XMLParser/XMLParser.cpp:29]: This is only for demo
Logger in new branch is using separate parser

File: XMLParser/XMLParser.h (hunk 6 above)
Comment (Hipst0man) [reply XMLParser/XMLParser.h:97]: This is only for demo
Logger in new branch is using separate parser

**Pull request closed without merging**